src/main/java/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java

```diff
@@ -191,23 +194,27 @@ public void registerRepo(String userId, RepoProvider repoDetails)
                         "] -> [" + newUrl + "]");
         }
         final RepoData repoData = JsonRepoProvider.repoProviderToData(this, repoDetails);
-        final DBCollection hist = db.getCollection(TABLE_REPO_HISTORY);
-        hist.insert(new BasicDBObject(FIELD_RH_MODULE_NAME, repoModuleName)
+        final MongoCollection<Document> hist = db.getCollection(TABLE_REPO_HISTORY);
+        hist.insertOne(new Document()
+                .append(FIELD_RH_MODULE_NAME, repoModuleName)
                 .append(FIELD_RH_VERSION, newVersion)
-                .append(FIELD_RH_REPO_DATA, toDBObject(repoData)));
+                .append(FIELD_RH_REPO_DATA, toDocument(repoData)));
         
-        final DBCollection data = db.getCollection(TABLE_REPO_INFO);
+        final MongoCollection<Document> data = db.getCollection(TABLE_REPO_INFO);
         //should just do an upsert rather than handling the logic application side
         if (wasReg) {
-            final Map<String, Object> info = toMap(data.findOne(
-                    new BasicDBObject(FIELD_RI_MODULE_NAME, repoModuleName)));
-            info.put(FIELD_RI_LAST_VERSION, newVersion);
-            info.put(FIELD_RI_STATE, RepoState.ready.toString());
-            // this is race condition city
```

Comment:
Let's keep this comment

Reply:
Why is this still a race condition even when we use updateOne?

Reply:
Because the db record can be changed between pulling the record from the db, altering it, and pushing it back

Reply:
Updated.

If we don't pull, then it should not be a race condition.

Reply:
:+1: